src/Framework/Interaction/InitialState.cxx

```diff
@@ -346,9 +346,9 @@ TLorentzVector * InitialState::GetProbeP4(RefFrame_t ref_frame) const
              // make sure that 'struck nucleon' properties were set in
              // the nuclear target object
 
-             assert( fTgt->HitNucP4Ptr() != 0 );
+	     assert( pdg::IsNucleon(fTgt->HitPartPdg()) );
 
-             TLorentzVector * pnuc4 = fTgt->HitNucP4Ptr();
+             TLorentzVector * pnuc4 = fTgt->HitPartP4Ptr();
 
              // compute velocity vector (px/E, py/E, pz/E)
 
```

Comment:
I believe we're missing a reference frame case for striking the electron's reference frame. https://github.com/bear-is-asleep/Generator/blob/255d21798dc6f7382ebe503015535326289fb272/src/Framework/Interaction/InitialState.cxx#L394-L410

Reply:
same comment for both `GetTgtP4` and `GetProbeP4` https://github.com/bear-is-asleep/Generator/blob/25410911ff9227f5b9eb05a6dc755aad892cb327/src/Framework/Interaction/InitialState.cxx#L322-L339